### PR TITLE
Fix depwarn on 0.5

### PR DESCRIPTION
**Pull request by yuyichao**
Requires JuliaLang/Compat.jl#192 and a new tag.

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,4 +1,4 @@
 julia 0.3
 BinDeps
-Compat
+Compat 0.7.14+
```
Comment (Luthaf) [inline REQUIRE]: Should'nt that line be `Compat 0.7.14` ? The Travis failure seems related to that.

**Comment by Luthaf** [comment]
Travis failure is related to the fact that the Compat changes did not land yet, I am correct?

**Comment by yuyichao** [comment]
Right. This should not be merged until the linked Compat PR is merged and a new version is tagged. The Compat version in `REQUIRE` is now a place holder and needs to be changed to the actual version when the Compat tag happends.

**Comment by yuyichao** [comment]
Updated.

Potential breaking changes:
https://github.com/Luthaf/Conda.jl/pull/31/files#diff-b5a035973669724cfc435a3914eae6ebL170 (if someone depends on the type of the return value)
https://github.com/Luthaf/Conda.jl/pull/31/files#diff-b5a035973669724cfc435a3914eae6ebL222 (if someone depends on the type of the return value)
https://github.com/Luthaf/Conda.jl/pull/31/files#diff-98af5ab944fe38d087a66203b1868a0dL5 (unlikely, only if someone read the field or depend on the aliasing of the arrays.)

**Comment by yuyichao** [comment]
Updated to take advantage of JuliaLang/Compat.jl#198. This doesn't require a new Compat version to work but does require a new tag to get rid of the depwarns on 0.5.

**Comment by Luthaf** [comment]
Thank you for this PR!

I did not follow the recent changes in Julia, the String type is now always UTF-8?

**Comment by yuyichao** [comment]
> the String type is now always UTF-8?

`ASCIIString` is gone.
String literals are always (UTF8)`String`
`UTF8String` is renamed to `String`.

**Pull request merged**